internal/kafka/test/common/util.go

```diff
@@ -239,23 +239,26 @@ func CheckMetricExposed(h *test.Helper, t *testing.T, metric string) {
 func IsMetricExposedWithValue(h *test.Helper, t *testing.T, metric string, values ...string) bool {
 	resp := getMetrics(t)
 	metricLines := strings.Split(resp, "\n")
-	metricValuesFound := false
-	for _, l := range metricLines {
-		if strings.Contains(l, metric) {
-			valuesFound := 0
-			for _, v := range values {
-				if !strings.Contains(l, v) {
-					valuesFound++
-				} else {
-					break
-				}
-			}
-			if len(values) == valuesFound {
-				metricValuesFound = true
+	for _, metricLine := range metricLines {
+		if !strings.Contains(metricLine, metric) {
+			continue
+		}
+
+		allValuesFound := true
+		for _, value := range values {
+			if !strings.Contains(metricLine, value) {
+				allValuesFound = false
+				break
 			}
 		}
+
+		if allValuesFound {
+			return true
+		}
```

Comment:
```suggestion
		if arrays.FindFirstString(values, func(value string) bool {
			return !strings.Contains(metricLine, value)
		}) != -1 {
			return true
		}
```

Reply:
Thanks. I've updated to use the suggested utility